### PR TITLE
feat: removed wallet_getCapabilities

### LINKED ISSUE #99: Remove `wallet_getCapabilities`
This is no longer used, so we can simplify quite a bit by just removing it. The only hurdle here is that we use it in e2e tests to get the delegation contract - however, any contract here will do, so we just need to adjust the tests to take e.g. an a contract address as an env var, or hardcode it.

### PR DESCRIPTION
**Pull request by varun-doshi**
Closes #99 

Removed `get_capabilities` from the `WalletCapabilities` trait
In `e2e_tests` the capabilities is initialised manually with the contract address being pulled in from `env` variables.

### CODE REVIEW

**Comment by varun-doshi** [comment]
There are 2 ways we can set the address in the capabilities var:
- Hardcode it to some contract address
- Pull from `std::env`

File: crates/e2e-tests/src/tests.rs [hunk 1]
```diff
@@ -54,11 +55,25 @@ async fn test_wallet_api() -> Result<(), Box<dyn std::error::Error>> {
...
+        );
+
+        let mut inner_tree = BTreeMap::new();
+        inner_tree.insert("delegation".to_string(), innermost_tree);
+
+        let mut capabilities = BTreeMap::new();
+        capabilities.insert(chain_id, inner_tree);
+
+        capabilities
+    };
```
Comment (onbjerg) [inline crates/e2e-tests/src/tests.rs:74]: we should remove the capabilities structs entirely, we don't need them anymore, and instead use a const address here or an env var

**Review by onbjerg — CHANGES_REQUESTED** [review]

File: crates/e2e-tests/src/tests.rs (hunk 1 above)
Comment (varun-doshi) [reply crates/e2e-tests/src/tests.rs:74]: Just to clarify, you mean to remove `WalletCapabilities` right?

```
#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
pub struct WalletCapabilities(pub HashMap<U64, Capabilities>);
```


I see its not really being used anywhere else in the code

Comment (onbjerg) [reply crates/e2e-tests/src/tests.rs:74]: yes

**Comment by onbjerg** [comment]
@varun-doshi looks good overall, some lints are failing PTAL

**Review by onbjerg — CHANGES_REQUESTED** [review]
sorry about this, can you resolve the merge conflicts as well? I can't push to your PR since you opened it from your main branch

**Review by onbjerg — CHANGES_REQUESTED** [review]
still some failing lints, maybe due to merge? sorry about this and thank you for your patience 🥲 if you open up your PR from another branch than main next time I can push to your pr to handle the nits

**Comment by varun-doshi** [comment]
> still some failing lints, maybe due to merge? sorry about this and thank you for your patience 🥲 if you open up your PR from another branch than main next time I can push to your pr to handle the nits

Understood

**Comment by varun-doshi** [comment]
Opened new PR with correct branch https://github.com/ithacaxyz/odyssey/pull/110
This can be closed

**Pull request closed without merging**